Add container attached service implementation

s223880826 opened the pull request:
This is the new service I got, named "Container Attached". This service had only one resource, and I got 4 policies from this.

christoraju085 committed:
Add container attached service implementation

jashan-lefty committed:
Linters script (#160)

* first commit

* update the inputs folder logic

Seb-sketch committed:
Update helpers.rego (#166)

Pat-Stuart committed:
Revert "Update helpers.rego (#166)" (#167)

This reverts commit 6af54154b58408d2a7c72b95691081f435c17ce8.

jashan-lefty commented on policies/gcp/container_attached/container_attached_cluster/oidc_issuer_suffix_whitelist/policy.rego:
```diff
@@ -0,0 +1,23 @@
+      "condition": "OIDC issuer must end with approved suffix",
+      "attribute_path": ["oidc_config", 0 , "issuer_url"],
+      "values": [".oidc.corp.example.com"],
+      "policy_type": "suffix_whitelist"
```
No policy type such as suffix whitelist

jashan-lefty commented on inputs/gcp/container_attached/container_attached_cluster/distribution_whitelist/c.tf:
```diff
@@ -0,0 +1,16 @@
+}
+
+resource "google_container_attached_cluster" "c" {
+  name         = "ac-oidc-ok"
```
Make sure the naming of the compliants and c and nc follow the strucuture

Complainatns- c
Non compliants - nc

Apply this to all configs

jashan-lefty reviewed (changes requested):
Hi mate, I see there are compilation errors in your Terraform, I should be due to incorrect terraform code as folows:

<img width="1081" height="490" alt="Image" src="https://github.com/user-attachments/assets/e47e9d5f-97aa-4156-b997-a6b258757136" />

Make sure the naming of the compliants and c and nc follow the strucuture

Complainatns- c
Non compliants - nc

Apply this to all configs

sumedh004 committed:
Gcp/service/cloud storage (#54)

* gcp cloud service storage

* updated asked changes

* asked changes by yotam

* naming cleanup

* naming cleanup

* default object access control resource

* policies for some resources

* add remaining policies

* reviewed existing inherited policies

* minor format

* address pr comments

* combine some policies

* naming cleanup

* various bug fixes and renaming for testing

* merge commit

* folders rename

* terraform fmt

* add documentation

* folder rename and documentation added

* doc generated

* slight update to docs

* slight update to docs

---------

Co-authored-by: AkashreddyVootk <s224127166@deakin.edu.au>

Vamsi57366 committed:
Finished contributing for Application Integration (#105)

* Restored deleted folders and applied review feedback

* Addressed feedback and added documentation

vinvincodes committed:
Gcp/service/cloud platform (#106)

* Work on cloud platform service

* add inputs & policies of cloud_platform

* Delete .vscode/settings.json

* delete template comments from all policy files

* undo changes to templates

* delete comput and storage folders

* update

* fix: applied contributor feedback

* Save local changes before merging

* update to name for  resource_value_name on vars.rego

* remove exempt_members policy due to doesn't work and helper.rego hasn't update for this policy

* Add/update JSON documentation for GCP resources - cloud platform

* Add/update markdown documentation for GCP resources - cloud platform

* all updated

* readme.md back

* Fix: resolved merge conflict in cloudfunctions2_function.json, removed markers and applied origin/dev security rules

* Fix: resolved merge conflict in cloudfunctions2_function_iam.json, removed markers and applied origin/dev security rules

* Fix: resolved merge conflict in model_armor_template.json & _floorsetting.json, removed markers and applied origin/dev security rules

* google_project update name

* update

* update google_service_account_key/exposure

* update google_folder_organization_policy/constraint

AK199417 committed:
Gcp/service/google cloud netapp volumes (#74)

Pat-Stuart committed:
Testing Script and CI Pipeline (#117)

* finished script and basic pipeline

* remove comment

* finished script and basic pipeline

* remove comment

jashan-lefty committed:
linter minor update (#173)

* first commit

* update the inputs folder logic

* Fix: Root path error fix (Alloes to run linters from any path) but it must me called from the root directory "Policy-Deployment-Engine

jashan-lefty committed:
Cloud key management CleanUP from T1 (#164)

* Add policies and inputs for Cloud Key Management

* Updated helpers

* updating

* Updating the policies

* final updates

* Changes after the review

* Initial changes

* initial changs

* vars repositioning

* Changes in naming

* final changes into, updating according to linters and pipeline

* final commits

* updated gitignore

* added readme

* update vars

Pat-Stuart committed:
clean up (#151)

sennan21 committed:
feat(looker-core): add 11 Looker Core security policies with documentation (#172)

- Added 11 Looker Core security policies with proper OPA rules
- Fixed duplicate variable declarations by creating separate variables.tf files
- Included comprehensive documentation with security annotations
- All policies pass terraform init/validate without errors
- Using proper service branch structure as requested

sumedh004 committed:
restore policy template (#182)

Good catch sumedh

christoraju085 committed:
Accomodated the changes

Miguwell-I committed:
Created: gcp/service/privileged access manager (#112)

* Initial Commit

* Updated privileged access manager entitlement policies

* added changes

* small change

* remove policy

christoraju085 committed:
Resolve both-added by taking remote
Al commit

Amended the changes

sumedh004 committed:
Feature/folder generator (#183)

* removed selenium scrapping

* remove cache json for service to use runtime cache

* create readme

* create readme

shivam3175 committed:
Finished policies on Google Lustre service (#114)

* Initial commit

* Fixed review comments

* Add all plan.json files

* Pushing Lustre branch

* Updated few files

* Added documentation

* Updated changes

* Updated

* Removed chronicle inputs and policy folder

Sundi202 committed:
GKE Hub Policies  (#107)

* GKE Hub Feature (Fleet Observability) -  require default logs routing mode COPY or MOVE (updated)

* Security policies for google_gke_hub_feature

* no_public_principals_binding policy for IAM binding

* no_public_principals_member policy for IAM member

* configmanagement security policies (git_secure_auth, git_approved_HTTPS, git_approved_dir, pc_enable_required)

* added policy_binding policy & authority_issuer policy

* GKEHub: remove deprecated google_gke_hub_feature_membership/git_approved_dir

* GKEHub: standardize resource_value_name to {c|nc}; fix authority_issuer; tighten RBAC/HTTPS controls

shivam3175 committed:
Finished 10 Policies on Chronicle Service (#82)

christoraju085 committed:
Resolve both-added by taking remote

christoraju085 committed:
Resolved and amended the queries

Seb-sketch committed:
Revert "feat(looker-core): add 11 Looker Core security policies with document…" (#189)

This reverts commit d1b10d5e8381f4d68a3420d40deffe39b80ddf73.

Seb-sketch committed:
Gcp/service/cloud_deploy (Old PR from forked repo not cloned branch) (#186)

* GKE Hub Policies  (#107)

* GKE Hub Feature (Fleet Observability) -  require default logs routing mode COPY or MOVE (updated)

* Security policies for google_gke_hub_feature

* no_public_principals_binding policy for IAM binding

* no_public_principals_member policy for IAM member

* configmanagement security policies (git_secure_auth, git_approved_HTTPS, git_approved_dir, pc_enable_required)

* added policy_binding policy & authority_issuer policy

* GKEHub: remove deprecated google_gke_hub_feature_membership/git_approved_dir

* GKEHub: standardize resource_value_name to {c|nc}; fix authority_issuer; tighten RBAC/HTTPS controls

* Policies and documentation

* helpers comment

* Policies and terraform files from older branch

* Required changes to helpers needed for policies

* Fix wrong edit

* Fixing naming for input files to be exact matches "c" or "nc"

* Missed name from last commit, also formatting plan files

* Changing name of folders to more descriptive label

---------

Co-authored-by: Sebastian <sebastianedge50@gmail.com>

HarshPandya7 committed:
GDCE Service Completed  (#98)

* Updated GDCE policies and input files for cluster security compliance

* Updated GDCE policies, inputs, and documentation; added Cloud Stackdriver Logging service

* "Updated Document of GDCE"

* feat: add GDCE plan.json outputs only

* Fixed GDCE policies: corrected node pool locations, KMS key validation, cluster target version, and maintenance policy checks. Removed CloudStackdriver logging service.

* Remove Cloud Stackdriver Logging service from inputs and policies

* added GDCE cluster CIDR blocks policy

patrickandersondeakin committed:
API Hub pull request (#93)

* initial policy for dlp

* temporary helpers.rego fix

* polciies for network resource

* polciies for network resource

* policies for network peering resource

* policies for network policy resource

* policies for external access rule

* merged helper rego

* policy rewrite for new helper

* fix policy

* policy for external access rule

* remove project from config

* restructure and minor changes

* folder cleanup

* Add API Hub policies and inputs for peer review.

* Checking policies and code

* API Hub pull request

* Changes made according to Pat Stewart PR

* Added some more changes according to Pat Stuart PR

* Markdown created for peer review

* Peer review changes made for Api_Hub

* Made two minor changes

---------

Co-authored-by: Sumedh Vartak <67580847+sumedh004@users.noreply.github.com>

LegendaryMercury committed:
Discovery engine PR Request (#81)

* Fixed Mistakes

* Made Some Compliancy Changes

* Made Some Requested Changes

Made some Changes

* Added Plan Files

Added Plan Files

* Pattern_whitelist

Changed 2 policys to pattern whitelist

* Changed_engine_schema_json

PatternWhitelist

* Updated_recommendation_engine

Removed unneeded stuff.

* Taking out Trash

Gone
(the location policys that arent in AUS)
(also file adjustments for the engine_schema json check, back to whitelist but with full expression)

* More Changes

More Changes Made

* Added location Policy's Back

Yotam Said i could add them, because i did a few they will just be worth less.

* misc

some misc changes.

* Documentation

Hope this is what you wanted.

* search_engine_fix

* engine_config_fix

* Opps

Opps

---------

Co-authored-by: Pat Stuart [SSW] <71543754+Pat-Stuart@users.noreply.github.com>

Chhunly-TAING committed:
Gcp/service/bigquery data transfer (#121)

* Added BigQuery Data transfer policies

* implement BigQuery Data Transfer policy

* I have made some changes to my c.tf and nc.tf

* Address the feedback

* Push with the documentation

* I have fixed the output of the opa eval

* I have deleted sensitive_params

* I am sorry Sumedh, I forgotten to delete the document fro mthe GitHub

Prem-CyberSec committed:
Gcp/service/os config v2 (#195)

* Restarted the branch due to conflicts with files for other services

* Added a documentation in the README for understanding the first policy deployment

* Added policy for the new resource Policy_Orchestrator_for_folder

This policy works mostly the same as the one seen in other resources, this is covering the attribute orchestrated_resource

* Added another policy for policy_orchestrator_for_folder

ACTION policy is created for the policy_orchestrator_for_folder

* Added a new policy for policy_orchestrator_for_folder

Policy added for orchestration scope which specifies the location

* Reverted README.md to original file matching from DEV

* Added Documentations required for the service

* PATCH FOR THE ORIGINAL FILES

* modified:   inputs/gcp/os_config_v2/policy_orchestrator/action/nc.tf

---------

Co-authored-by: Seb-sketch <SebastianE50@pm.me>

visal-dam committed:
Created helpers.md (#196)

Technical documentation for the helpers.rego for future developers and maintainers.

MenuraDewalegama committed:
Gcp/service/bigquery data policy (#63)

* adding bigquery_data policies

* fixing indentation issues

* removing NSR policies

* fixing the vars.rego file name

* whitelisting all the Australian zones

* chnaging the value of resource_value_name

* expanding the policy to blacklist against allUsers and other conditions

* adding documentation

waffa-alhasani committed:
completed gcp Firebase service (#76)

* completed gcp Firebase service

* completed gcp Firebase service

sennan21 committed:
fix(looker_core) rename input resource labels to c/nc (#193)

* fix(looker_core): rename input resource labels to c/nc; regenerate lockfiles

* fix(looker_core): change name attribute from ok/bad to c/nc in c.tf & nc.tf (per PR #193 review)

* fix(looker_core): update name attributes to cc/nc and regenerate plan.json files (per PR #193 review)

* fix(looker_core): update name attributes to c1/nc per vars.rego specification and regenerate plan.json files

* fix(looker_core): set name values to c1/nc1 per reviewer guidance

* fix(looker_core): ensure c1/nc1 naming is consistent across all policies (per PR #193 review)

jashan-lefty commented:
Hi Bro there are conflicts in your branch and the main branch, if I merge you branch, it will change all the files in the main branch. Make sure that you are only touching the folder under your service. Could you please resolve the new branches or submit a new clean PR?

sennan21 committed:
Gcp/service/recaptcha enterprise docs (#192)

* feat(gcp/service/recaptcha_enterprise): add allow_all_domains policy + vars + c/nc + config.tf (config unchanged)

* chore(recaptcha_enterprise): move service-level vars.rego to key/

* GCP reCAPTCHA Enterprise key: add policies for allow_amp_traffic, challenge_security_preference, integration_type (inputs + policy)

* docs(recaptchaenterprise): annotate web_settings security fields and generate markdown

* recaptchaenterprise/key: remove duplicate vars; add test evidence (.terraform.lock.hcl) per input; confirm c/nc config present

chunyiw8080 committed:
update name attribute values and docs for memcache (#159) (#194)

* update name attribute values and docs for memcache

* remove empty array terraform nc example

---------

Co-authored-by: chunyi wang <whunyi@foxmail.com>

Monneh committed:
dataform policies  (#197)

* feat: add dataform policies and inputs from gcp/service/dataforms branch

* feat: Add comprehensive Dataform service with 4 resources and documentation

- Add 4 Dataform resource types with complete policy coverage:
  - google_dataform_repository (8 policies)
  - google_dataform_repository_iam (1 policy)
  - google_dataform_repository_release_config (1 policy)
  - google_dataform_repository_workflow_config (1 policy)

- Add comprehensive security policies covering:
  - Deletion policy enforcement (FORCE vs DELETE)
  - Region allowlist restrictions
  - CMEK encryption requirements
  - Git authentication security (HTTPS/SSH)
  - Required security labels
  - IAM public access prevention
  - Cron schedule requirements
  - Service account enforcement

- Add complete documentation with security impact analysis:
  - 4 JSON resource documentation files with security rationales
  - 4 generated Markdown documentation files
  - Compliant/non-compliant examples for all security-critical arguments

- Add comprehensive test inputs:
  - Compliant (c.tf) and non-compliant (nc.tf) examples
  - Updated plan.json files for policy testing
  - Complete variable and configuration files

All policies tested and validated with OPA evaluation.

s223880826 committed:
updating the policies with changes

s223880826 committed:
Updated the changes

s223880826 committed:
updating the policies with changes

s223880826 committed:
Updated the changes

christoraju085 committed:
Latest updated policies

s223880826 committed:
Add files via upload

s223880826 committed:
Add files via upload

s223880826 committed:
Updating the changes

s223880826 committed:
Updated the changes on the policy Container Attached

s223880826 committed:
Updated the changes

s223880826 committed:
Updated the changes on the policy Container Attached

s223880826 committed:
Container attached latest

s223880826 committed:
Container attached latest

s223880826 committed:
Container attached latest

jashan-lefty commented:
Bro you branch seems to have conflict errors with all the other files. Could you please restore the previous version of your branch to solve the conflicts?
